src/contracts/Yieldy.sol

```diff
@@ -87,30 +76,28 @@ contract Yieldy is
         @param _epoch uint256 - epoch number
      */
     function rebase(uint256 _profit, uint256 _epoch)
-        public
-        onlyStakingContract
+        external
+        onlyRole(REBASE_ROLE)
     {
-        uint256 circulatingSupply_ = circulatingSupply();
-        require(circulatingSupply_ > 0, "Can't rebase if not circulating");
-
-        uint256 rebaseAmount;
         uint256 currentTotalSupply = _totalSupply;
+        require(_totalSupply > 0, "Can't rebase if not circulating");
 
         if (_profit == 0) {
             emit LogSupply(_epoch, block.timestamp, currentTotalSupply);
             emit LogRebase(_epoch, 0, getIndex());
         } else {
-            rebaseAmount = (_profit * currentTotalSupply) / circulatingSupply_;
-            uint256 updatedTotalSupply = currentTotalSupply + rebaseAmount;
+            uint256 updatedTotalSupply = currentTotalSupply + _profit;
 
             if (updatedTotalSupply > MAX_SUPPLY) {
                 updatedTotalSupply = MAX_SUPPLY;
             }
 
-            gonsPerFragment = TOTAL_GONS / updatedTotalSupply;
+            rebasingCreditsPerToken = rebasingCredits / updatedTotalSupply;
+            require(rebasingCreditsPerToken > 0, "Invalid change in supply");
+
             _totalSupply = updatedTotalSupply;
```

Comment:
Ohhh, I was curious about this.  MNH didn't like us overriding the totalSupply and I was confused how we could mint for rebases.  It makes sense we would still need to do this